spec/index.bs

```diff
@@ -251,6 +251,157 @@ dictionary GPUObjectDescriptorBase {
+  2. User issues {{GPUQueue/submit(commandBuffers)|GPUQueue.submit()}} that hands over
+    the {{GPUCommandBuffer}} to the user agent, which processes it
+    on the [=Device timeline=] by calling the OS driver to do a low-level submission.
+  3. The submit gets dispatched by the GPU thread scheduler onto the
+    actual compute units for execution, which happens on the [=Queue timeline=].
+
+</div>
+<div class="example">
+{{GPUDevice/createBuffer(descriptor)|GPUDevice.createBuffer()}}:
+
+  1. User fills out a {{GPUBufferDescriptor}} and creates a {{GPUBuffer}} with it,
+    which happens on the [=Content timeline=].
+  2. User agent creates a low-level buffer on the [=Device timeline=].
+
+</div>
+<div class="example">
+{{GPUBuffer/mapReadAsync()|GPUBuffer.mapReadAsync()}}:
+
+  1. User requests to map a {{GPUBuffer}} on the [=Content timeline=],
+    gets a promise in return.
+  2. User agent checks if the buffer is currently used by the GPU
+    and makes are reminder to itself to check back when this usage is over.
+  3. After the GPU operating on [=Queue timeline=] is done using the buffer,
+    the user agent maps it to memory and resolves the promise.
+
+</div>
+
+## Memory ## {#programming-model-memory}
+
+*This section is non-normative.*
+
+Once a {{GPUDevice}} has been obtained during an application initialization routine,
+we can describe the <dfn dfn>WebGPU platform</dfn> as consisting of the following layers:
+  1. User agent implementing the specification.
+  2. Operating system with low-level native API drivers for this device.
+  3. Actual CPU and GPU hardware.
+
+Each layer of the [=WebGPU platform=] may have different memory types
+that the user agent needs to consider when implementing the specification:
+  - The script-owned memory, such as {{ArrayBuffer}} created by the script,
+    is generally not accessible by a GPU driver.
+  - A user agent may have different processes responsible for running
+    the content and communication to the GPU driver.
+    In this case, it uses inter-process shared memory to transfer data.
+  - Dedicated GPUs have their own memory of high bandwidth,
+    while integrated GPUs typically share memory with the system.
+
+Most [=physical resources=] are allocated in the memory of type
+that is efficient for computation or rendering by the GPU.
+When the user needs to provide new data to the GPU,
+the data may first need to cross the process boundary in order to reach
+the user agent part that communicates with the GPU driver.
+Then it may need to be made visible to the driver,
+which sometimes requires a copy into driver-allocated staging memory.
+Finally, it may need to be transferred to the dedicated GPU memory,
+potentially changing the internal layout into one
+that is most efficient for GPUs to operate on.
+
+All of these transitions are done by the WebGPU implementation of the user agent.
+
+Note: This example describes the worst case, while in practice
```

Comment:
nit: Since this starts with `Note:` it renders in a green box as a note, which I don't think is necessary. Maybe write as `(Note: ...)`?

Reply:
That was intentional, actually.

Reply:
OK, gotcha.